src/time/format/formatter.cr

```diff
@@ -186,12 +186,12 @@ struct Time::Format
       end
     end
 
-    def time_zone_offset(force_colon = false, allow_colon = true, allow_seconds = true)
-      time.zone.format(io, with_colon: force_colon, with_seconds: allow_seconds)
+    def time_zone_offset(force_colon = false, allow_colon = true, format_seconds = true, parse_seconds = true)
+      time.zone.format(io, with_colon: force_colon, with_seconds: format_seconds)
```

Comment:
`parse_seconds` isn't used in this method;

Reply:
Yes, but `Format::Parser.time_zone_offset` does and both types should have the same interface for `Time::Pattern` (even though it's currently not used anywhere).

Reply:
Still, the parser and formatter shouldn't have irrelevant parameters. Please keep `allow_seconds`.

Note that I'd drop support for seconds in timezone offsets altogether, since neither RFC 3339, RFC 2822 or ISO 8601 support it.

Reply:
The time zone database is based on seconds and so is `Time:: Location`. It should be possible to completely serialize it. Offset seconds are used by the format pattern `%::z`.

Reply:
I'm not sure what you mean by keeping `allow_seconds`. That's semantically the same as `parse_seconds`. But I think the latter name fits better with the new argument `write_seconds`.

Reply:
I mean: don't introduce new parameters that make no sense in some context (`parse_seconds` in formater?! `format_seconds` in parser?!). Keeping the existing and generic `allow_seconds` would fit in both context, and fit with other namings (such as `allow_colon`) too.

Reply:
We can't use the same name for different arguments. Both the formatter and parser implementations of the method can be called from the same pattern method (for example in `Pattern#rfc_3339`).

Reply:
So? If a pattern wants seconds in the parser, then it also wants seconds in the formatter.

Reply:
No, not necessarily. A format could leniently support parsing seconds but don't print them. That's essentially what I'd suggested for the RFC 3339 format.